extensions/outline-document-search/src/InstancesList.tsx

```diff
@@ -0,0 +1,189 @@
+
+const useInstances = () => {
+  const [instances, setInstances] = useState<Instance[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const fetchInstances = async () => {
+    setIsLoading(true);
+
+    const instances = await LocalStorage.getItem(localStorageKey);
+
+    if (instances) {
+      if (typeof instances === "string") {
+        setInstances(JSON.parse(instances));
+      }
+    } else {
+      setInstances([]);
+    }
+
+    setIsLoading(false);
+  };
+
+  const removeInstance = async (instance: Instance) => {
+    setIsLoading(true);
+
+    await LocalStorage.setItem(
+      localStorageKey,
+      JSON.stringify(instances.filter((existingInstance) => existingInstance !== instance)),
+    );
+    await fetchInstances();
+  };
+
+  useAsync(async () => {
+    await fetchInstances();
+  }, []);
+
+  return {
+    instances,
+    isLoading,
+    fetchInstances,
+    removeInstance,
+    setIsLoading,
+  };
+};
+
+const AddInstanceAction = ({
+  fetchInstances,
+  shortcut = undefined,
+  title = "Add Instance",
+}: {
+  fetchInstances: () => Promise<void>;
+  shortcut?: Shortcut;
+  title: string;
+}) => <Action.Push {...{ shortcut, title }} icon={Icon.Plus} target={<InstanceForm onAdd={fetchInstances} />} />;
+
+const InstancesList = () => {
+  const { instances, isLoading, fetchInstances, removeInstance, setIsLoading } = useInstances();
+
+  const InstancesActions = ({
+    shortcut = undefined,
+    title = "Add Instance",
+  }: {
+    shortcut?: Shortcut;
+    title?: string;
+  }) => (
+    <>
+      <AddInstanceAction {...{ fetchInstances, shortcut, title }} />
+      {instances.length > 0 && (
+        <Action
+          icon={Icon.Trash}
+          onAction={async () => {
+            setIsLoading(true);
+
+            await LocalStorage.removeItem(localStorageKey);
+            await fetchInstances();
+          }}
+          shortcut={{ key: "delete", modifiers: ["ctrl", "shift"] }}
+          style={Action.Style.Destructive}
+          title="Remove All"
+        />
+      )}
+    </>
+  );
+
+  useAsync(fetchInstances, []);
+
+  if (instances.length === 0) {
+    return (
+      <List
+        actions={
+          <ActionPanel>
+            <InstancesActions />
+          </ActionPanel>
+        }
+        isLoading={isLoading}
+        navigationTitle="Outline Instances"
+      >
+        <List.EmptyView
+          icon={{ source: "https://www.getoutline.com/images/logo.svg" }}
+          title="Start by adding an Outline instance..."
+        />
+      </List>
+    );
+  } else {
+    return (
+      <List isLoading={isLoading} navigationTitle="Outline Instances">
+        {instances.map((instance, index) => (
+          <InstanceItem
+            {...{ fetchInstances, instance }}
+            key={index}
+            remove={() => removeInstance(instance)}
+            instancesActions={<InstancesActions shortcut={{ key: "a", modifiers: ["cmd", "shift"] }} title="Add" />}
```

Comment:
Is there a sensible way to come up with these shortcuts?

Reply:
We have some default shortcuts in `Keyboard.Shortcut.Common.`